plugins/foreman_virt_who_configure/0.5/index.md

```diff
@@ -0,0 +1,272 @@
+
+```sh
+hammer virt-who-config fetch --id 4
+hammer virt-who-config fetch --id 4 --output /root/script.sh
+```
+
+The first example prints the configuration script on STDOUT, second stores it into the specified path. Hammer CLI also checks that the file does not exist previously to avoid a race condition security risk. It also configures permissions properly so only the current user can read and execute it. In both cases, `--name` could be used instead of `--id` 
+
+#### Viewing a config
+
+```sh
+$ hammer virt-who-config info --id 1
+
+General information: 
+    Id:                  1
+    Name:                VMware
+    Hypervisor type:     esx
+    Hypervisor server:   vcenter.example.com
+    Hypervisor username: username
+    Status:              OK
+Schedule:            
+    Interval:       every 4 hours
+    Last Report At: 2021/10/14 16:14:09
+Connection:          
+    Satellite server:     foreman.example.com
+    Hypervisor ID:        hostname
+    Filtering:            Unlimited
+    Filter host parents:  
+    Exclude host parents: 
+    Debug mode:           no
+    Ignore proxy:
+```
+
+#### Creating a config
+
+```sh
+hammer virt-who-config create --name="gCBHwsVPFZ" --debug="1" --interval="60" --hypervisor-id="hostname" --hypervisor-type="esx" --organization-id="1" --filtering-mode="none" --satellite-url="foreman.example.com" --hypervisor-server="esxi.example.com" --hypervisor-username="admin" --hypervisor-password="password"
+
+Virt Who configuration [gCBHwsVPFZ] created
+```
+
+#### Updating a config
+
+```sh
+$ hammer virt-who-config update --id 5 --hypervisor-password="changeme"
+
+Virt Who configuration [gCBHwsVPFZ] updated
+```
+
+#### Deleting a config
+
+```sh
+$ hammer virt-who-config delete --id 5
+
+Virt Who configuration deleted
+```
+
+## 3.3 Monitoring
+
+Once the virt-who configuration is successfully deployed, the virt-who daemon runs and periodically send reports about what VMs it found on the hypervisors. Every report upload updates the configuration in Foreman, so you can monitor the configuration status. If you navigate to virt-who configurations list page, one of the table columns is called __Status__. Each configuration can have one of the following status:  
+
+* **No Report Yet** - This means, no virt-who report was recieved by Foreman yet, either the configuration was not deployed yet or there was some error during the deployment or virt-who can't connect to Foreman
+* **OK** - We received the report within the scheduled interval, everything is OK
+* **No Change** - virt-who does not send any report when it does not find any change on hypervisor, if we don't receive any report during the scheduled interval, we consider the configuration as unchanged. This could also happen if virt-who fails to upload reports for whatever reason, if you know there was new VM added on the hypervisor but you see the configuration in __No Change__ state, Check that virt-who is running and its logs.
```

Comment:
```suggestion
* **No Change** - Virt-who does not send any report when it does not find any change on hypervisor. If Foreman doesn't receive any report during the scheduled interval, the configuration is considered unchanged. This could also happen if virt-who fails to upload reports for whatever reason. If you know there was a new VM added on the hypervisor but you see the configuration in __No Change__ state, check that virt-who is running and its logs.
```